builder/src/lib.rs

```diff
@@ -74,37 +73,24 @@ impl Service {
                 output_path.to_str().unwrap(),
                 path.to_str().unwrap(),
             ])
-            .stdout(Stdio::piped())
-            .stderr(Stdio::piped())
-            .spawn()?;
-
-        let stdout = child.stdout.take().expect("to get handle on stderr");
-        let stderr = child.stderr.take().expect("to get handle on stderr");
-        let mut stderr_reader = BufReader::new(stderr).lines();
-        let mut stdout_reader = BufReader::new(stdout).lines();
-        let mut stdout_ended = false;
-        let mut stderr_ended = false;
-        while !(stdout_ended && stderr_ended) {
-            tokio::select! {
-                stderr_line = stderr_reader.next_line() => {
-                    match stderr_line {
-                        Ok(Some(line)) => info!(deployment_id, "{line}"),
-                        Ok(None) => stderr_ended = true,
-                        Err(err) => {
-                            error!(deployment_id, "unexpected stderr stream close: {}", err);
-                            stderr_ended = true;
-                        }
-                    }
-                },
-                stdout_line = stdout_reader.next_line() => {
-                    match stdout_line {
-                        Ok(Some(line)) => info!(deployment_id, "{line}"),
-                        Ok(None) => stdout_ended = true,
-                        Err(err) => {
-                            error!(deployment_id, "unexpected stdout stream close: {}", err);
-                            stdout_ended = true;
-                        }
-                    }
+            .stdout(writer)
+            .stderr(writer_clone);
+        let mut child = command.spawn()?;
+
+        // Avoid a deadlock.
+        drop(command);
```

Comment:
How can this happen if we're not dropping here?

Reply:
It's documented in the os_pipe docs here: <https://docs.rs/os_pipe/latest/os_pipe/#examples>